lib/puppet/type/file.rb

```diff
@@ -393,6 +393,22 @@ def self.title_patterns
 
     self.warning "Checksum value is ignored unless content or source are specified" if self[:checksum_value] && !self[:content] && !self[:source]
 
+    if Puppet.features.microsoft_windows? && parameter(:source) &&
+      [:use, :use_when_creating].include?(self[:source_permissions])
+      err_msg = "Copying owner/mode/group from the source file on Windows" <<
+      " is not supported; use source_permissions => ignore."
+
+      if self[:owner] == nil || self[:owner] == :absent || self[:group] == nil ||
+        self[:group] == :absent || self[:mode] == nil || self[:mode] == :absent
```

Comment:
The only comment I saw was https://github.com/puppetlabs/puppet/blob/master/lib/puppet/type/file/mode.rb#L165-L166. This may not be necessary, but `source_permissions => use` will cause `:absent` to be replaced by the source. Maybe it makes sense to keep handling of `:absent` (skip updating it on Windows) in `source.rb`.